ts/src/encoding/fastPforEncodingUtils.spec.ts

```diff
@@ -0,0 +1,211 @@
+import { describe, expect, it } from "vitest";
+
+import { decodeFastPfor } from "../decoding/integerDecodingUtils";
+import IntWrapper from "../decoding/intWrapper";
+import { encodeFastPfor } from "./integerEncodingUtils";
+import { bigEndianBytesToInt32s } from "../decoding/byteIO";
+
+const BLOCK = 256;
+
+/**
+ * Reads a signed int32 value from a big-endian byte buffer.
+ *
+ * @param buf - Source byte buffer.
+ * @param offset - Byte offset.
+ * @returns The decoded int32 value.
+ */
+function readInt32BigEndian(buf: Uint8Array, offset: number): number {
```

Comment:
Move these methods to a different file (bytesIOencoder or something If I understand what they do), use them in a separate unit test files.